financial-templates-lib/logger/Logger.js

```diff
@@ -30,6 +30,12 @@
 const winston = require("winston");
 const { transports } = require("./Transports");
 
+async function waitForLogger(logger) {
+  const loggerDone = new Promise(resolve => logger.on("finish", resolve));
+  logger.end();
+  return loggerDone;
+}
```

Comment:
Shouldn't this return `await loggerDone`? Otherwise won't the result be a promise wrapped in another promise? I could be wrong here, but my understanding was that the return values of async functions were always wrapped in promises.

Reply:
This is meant to return a promise. The function `waitForLogger` is called as an async from the bot itself. So the bot calls `waitForLogger(Logger)` which returns a promise. When that resolves the flow continues and the process ends. I can try adding an additional `await` here but this is not what the reference implementation did and this script seems to work great from my testing.

Reply:
My understanding is that async functions wrap their return values in promises already, so `waitForLogger` would return a promise even if it returned `await loggerDone` I think.

Could you try one or both of these options to see if they break it?
- Change the return statement to: `return await loggerDone;`
- Remove the async in front of `function waitForLogger(logger) {`

Reply:
I tested both of your options and it works correctly in both cases.

Reply:
wdyt about using one of these constructions to avoid the double wrapping?

Reply:
have implemented option 1.